lib/train/transports/cisco_ios.rb

```diff
@@ -0,0 +1,147 @@
+# encoding: utf-8
+
+require 'train/plugins'
+require 'train/transports/ssh'
+
+module Train::Transports
+  class BadEnablePassword < Train::TransportError; end
+
+  class CiscoIOS < SSH
+    name 'cisco_ios'
+
+    option :host, required: true
+    option :user, required: true
+    option :port, default: 22, required: true
+
+    option :password, required: true
+
+    # Used to elevate to enable mode (similar to `sudo su` in Linux)
+    option :enable_password
+
+    def connection
+      validate_options(@options)
+      @connection ||= Connection.new(@options)
+    end
+
+    class Connection < BaseConnection
+      def initialize(options)
+        super(options)
+
+        @host = @options.delete(:host)
+        @user = @options.delete(:user)
+        @port = @options.delete(:port)
+
+        @enable_password = @options.delete(:enable_password)
+
+        @prompt = /^\S+[>#]\r\n.*$/
+      end
+
+      def uri
+        "ssh://#{@user}@#{@host}:#{@port}"
+      end
+
+      private
+
+      def establish_connection
+        logger.debug("[SSH] opening connection to #{self}")
+
+        @ssh = Net::SSH.start(
+          @host,
+          @user,
+          @options.delete_if { |_key, value| value.nil? },
```

Comment:
Previously this was `@options.clone.delete_if { |_key, value| value.nil? }` and I presumed the intent was to avoid mutating `@options` even if you don't want the entries with nil values for this statement, and that's why I suggested `@options.reject { |_, value| value.nil? }` (which doesn't mutate anything at all) instead.

So I just wanted to make sure there wasn't confusion here. Maybe it's OK to strip the empty pairs here, but it seems like a thing that could cause trouble if you're not expecting it elsewhere.

Reply:
Nope, I see what you mean now, I'll go with `reject` here.